lib/utils/log/writer.go

```diff
@@ -43,3 +50,85 @@ func (s *SharedWriter) Write(p []byte) (int, error) {
 func NewSharedWriter(w io.Writer) *SharedWriter {
 	return &SharedWriter{Writer: w}
 }
+
+// FileSharedWriter is similar to SharedWriter except that it requires a os.File instead of a io.Writer.
+// This is to allow the File reopen required by logrotate and similar tools.
+// SharedWriter must be used for log destinations that don't have the reopen requirement, like stdout and stderr.
+// This is thread safe.
+type FileSharedWriter struct {
+	*os.File
```

Comment:
Avoid embedding `*os.File`, it's never safe to do anything with it unless you are the `FileSharedWriter` anyway.

You might get away with not using a lock (which is always sketchy, when doing I/O) if you use an `atomic.Pointer[os.File]` here instead.